api/cron/refreshLeaderboard.ts

```diff
@@ -3,26 +3,69 @@ import { gql, request } from "graphql-request";
 import { getModel } from "../../utils/mongo";
 import { TRADING_COMPETITION_SUBGRAPH } from "../../utils";
 
-const refreshTradingCompLeaderboard = async () => {
-  console.log("TradingCompLeaderboard refresh begin");
+const teamIds = ["1", "2", "3"] as const;
+
+type TeamId = typeof teamIds[number];
+
+type User = {
+  id: string;
+  team: { id: string };
+  volumeUSD: string;
+  globalRank?: number;
+};
+
+const teamResult: Record<TeamId, User[]> = {
+  "1": [],
+  "2": [],
+  "3": [],
+};
+
+const combineTeamResult = () => {
+  return [...teamResult["1"], ...teamResult["2"], ...teamResult["3"]].sort((a, b) => {
+    return Number(a.volumeUSD) > Number(b.volumeUSD) ? -1 : 1;
+  });
+};
+
+const refreshTradingCompLeaderboard = async (team: TeamId = "1", skip = 0) => {
+  console.log("TradingCompLeaderboard refresh begin team:", team, "skip:", skip);
 
   const { users } = await request(
     TRADING_COMPETITION_SUBGRAPH,
     gql`
-      {
-        users(orderBy: volumeUSD, orderDirection: desc) {
+      query getUsersQuery($team: String, $skip: Int) {
+        users(
+          orderBy: volumeUSD
+          orderDirection: desc
+          first: 1000
+          skip: $skip
+          where: { team: $team }
+        ) {
           id
           volumeUSD
           team {
             id
           }
         }
       }
-    `
+    `,
+    {
+      team,
+      skip,
+    }
   );
 
   console.log("Fetched users count: {}", users.length);
+  if (users.length > 0) {
+    teamResult[team].push(users);
+    if (skip + 1000 < 6000) {
+      await refreshTradingCompLeaderboard(team, skip + 1000);
+    }
+  }
+
+  console.log("TradingCompLeaderboard refresh ended");
```

Comment:
Maybe worth changing it to
```js
console.log("TradingCompLeaderboard refresh begin team:", team, "skip:", skip);
```
and bringing `console.log("TradingCompLeaderboard refresh ended");` down to the end of `updateLeaderboard` just in case we'll need to study the logs again 👀